spec/server/functional/subscription.spec.coffee

```diff
@@ -1638,6 +1654,22 @@ describe 'POST /paypal/webhook', ->
               payments = yield Payment.find({'payPalSale.id': @paymentEventData.resource.id}).lean()
               expect(payments?.length).toEqual(1)
 
+          describe 'when initial subscribe payment already recorded', ->
+            beforeEach utils.wrap ->
+              yield utils.clearModels([Payment])
+              @payment = yield Payment({'payPalBillingAgreementID': @paymentEventData.resource.billing_agreement_id}).save()
+              payments = yield Payment.find({'payPalBillingAgreementID': @paymentEventData.resource.billing_agreement_id}).lean()
+              expect(payments?.length).toEqual(1)
+
+            it 'does not create a new payment and updates the existing one', utils.wrap ->
+              url = getURL('/paypal/webhook')
+              [res, body] = yield request.postAsync({ uri: url, json: @paymentEventData })
+              expect(res.statusCode).toEqual(200)
+              expect(res.body).toEqual("Payment sale object #{@paymentEventData.resource.id} added to initial payment #{@payment.id}")
+              payments = yield Payment.find({'payPalSale.id': @paymentEventData.resource.id}).lean()
+              expect(payments?.length).toEqual(1)
+              expect(payments[0].payPalBillingAgreementID).toEqual(@paymentEventData.resource.billing_agreement_id)
+
```

Comment:
This should maybe also test the second payment on the webhook. Does it create a second payment?